src/perception/cost_map/cost_map.processing.cpp

```diff
@@ -10,39 +66,36 @@ namespace mrover {
         if (!mPublishCostMap) return;
 
         try {
-            SE3d zed_to_map = SE3Conversions::fromTfTree(mTfBuffer, "zed_left_camera_frame", "map");
+            SE3f cameraToMap = SE3Conversions::fromTfTree(mTfBuffer, "zed2i_left_camera_frame", "map").cast<float>();
             auto* points = reinterpret_cast<Point const*>(msg->data.data());
-            // std::for_each(points, points + msg->width * msg->height, [&](auto* point) {
-            for (auto point = points; point - points < msg->width * msg->height; point += mDownSamplingFactor) {
-                Eigen::Vector4d p{point->x, point->y, point->z, 1};
-                point_matrix.col((point - points) / mDownSamplingFactor) = p;
-                Eigen::Vector4d normal{point->normal_x, point->normal_y, point->normal_z, 0};
-                normal_matrix.col((point - points) / mDownSamplingFactor) = normal;
+            for (std::size_t i = 0, r = 0; r < msg->height; r += mDownSamplingFactor) {
+                for (std::size_t c = 0; c < msg->width; c += mDownSamplingFactor) {
+                    auto* point = points + r * msg->width + c;
+                    Eigen::Vector4f pointInCamera{point->x, point->y, point->z, 1};
+                    Eigen::Vector4f normalInCamera{point->normal_x, point->normal_y, point->normal_z, 0};
+
+                    mPointsInMap.col(static_cast<Eigen::Index>(i)) = cameraToMap.transform() * pointInCamera;
```

Comment:
Transform right in this for loop. May be slower though if Eigen optimizes batch multiply